File: frontend/src/types/role.ts
export interface Skill {
  id: string;
  slug: string;
  name: string;
  category: string;
  weight: number;
}

export interface Role {
  id: string;
  slug: string;
  name: string;
  description: string;
  skills: Skill[];
}

export interface RolesListResponse {
  roles: Role[];
}

export interface RoleSkillsResponse {
  role_id: string;
  role_name: string;
  skills: Skill[];
}

File: backend/app/schemas/role.py
"""Pydantic schemas for roles and skills."""

from uuid import UUID

from pydantic import BaseModel


class SkillResponse(BaseModel):
    id: UUID
    slug: str
    name: str
    category: str
    weight: int


class RoleResponse(BaseModel):
    id: UUID
    slug: str
    name: str
    description: str
    skills: list[SkillResponse]


class RolesListResponse(BaseModel):
    roles: list[RoleResponse]


class RoleSkillsResponse(BaseModel):
    role_id: UUID
    role_name: str
    skills: list[SkillResponse]
